pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -922,19 +930,68 @@ func (l *LibvirtDomainManager) SyncVMI(vmi *v1.VirtualMachineInstance, allowEmul
 		return nil, err
 	}
 
+	detachedDisks := getDetachedDisks(vmi.Name, vmi.Namespace, oldSpec.Devices.Disks, domain.Spec.Devices.Disks)
+	for attachDiskChannelName, attachDiskChannel := range l.attachDiskChannels {
+		if _, ok := detachedDisks[attachDiskChannelName]; !ok {
+			logger.V(hotplugDetachmentVerbosity).Infof("Hotplug detach stil requested, continue grace period = [%+v]", attachDiskChannelName)
+		} else {
+			// If disk is no longer in detachedDisk list, it no longer needs to be detached.  Signal
```

Comment:
this comment seems backwards ... or rather misplaced.

the grace period should be allowed to continue if `ok := detachedDisks[attachDiskChannelName]; ok` not `!ok` ... i.e. if the disk is still in the detachedDisk list.

if the disk is no longer in the detached list (`!ok`), then the channel should be signalled.

Reply:
better now.